### PR TITLE
CLBlast 1.3.0 (new formula)

### PR DESCRIPTION
**Pull request by CNugteren**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

A formula for [CLBlast: Tuned OpenCL BLAS library](https://github.com/CNugteren/CLBlast).

### CODE REVIEW

File: Formula/clblast.rb [hunk 1]
```diff
@@ -0,0 +1,16 @@
+class Clblast < Formula
+  desc "Tuned OpenCL BLAS library"
+  homepage "https://github.com/CNugteren/CLBlast"
+  url "https://github.com/CNugteren/CLBlast/archive/1.3.0.tar.gz"
+  sha256 "fd5e06c261551366b46ccbc7841012cc69b58c74ed641db4f7a157c055d24223"
+  depends_on "cmake" => :build
```
Comment (fxcoudert) [inline Formula/clblast.rb]: Blank line before and after `depends_on`

File: Formula/clblast.rb [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+  def install
+    system "cmake", ".", *std_cmake_args
+    system "make", "install"
+    pkgshare.install "samples/sgemm.c" # for a simple library linking test
+  end
+  test do
```
Comment (fxcoudert) [inline Formula/clblast.rb]: Blank line before `test do`

File: Formula/clblast.rb [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+    system ENV.cc, pkgshare/"sgemm.c", "-I#{include}", "-L#{lib}",
+                   "-lclblast", "-framework", "OpenCL"
```
Comment (fxcoudert) [inline Formula/clblast.rb]: Could we run the test, and check its output?

**Review by fxcoudert — CHANGES_REQUESTED** [review]
Simple and nice! That's how we love them 👍

Minor nit: the formula name in the commit message should be lowercase

File: Formula/clblast.rb (hunk 3 above)
Comment (CNugteren) [reply Formula/clblast.rb]: This is a sample program, so the 'test' here is to see if it compiles and links. CLBlast has a lot of real correctness tests, but they need to be explicitly compiled and run on an OpenCL device. From a Formula of a very similar library (clBLAS), I saw that [OpenCL doesn't work](https://github.com/Homebrew/homebrew-core/blob/7087358a3e8938a500bd7f8c67967821b58666d6/Formula/clblas.rb#L30) on the test systems?

File: Formula/clblast.rb (hunk 1 above)
Comment (CNugteren) [reply Formula/clblast.rb]: Done

File: Formula/clblast.rb (hunk 2 above)
Comment (CNugteren) [reply Formula/clblast.rb]: Done

**Review by fxcoudert — APPROVED** [review]

File: Formula/clblast.rb [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+  homepage "https://github.com/CNugteren/CLBlast"
+  url "https://github.com/CNugteren/CLBlast/archive/1.3.0.tar.gz"
+  sha256 "fd5e06c261551366b46ccbc7841012cc69b58c74ed641db4f7a157c055d24223"
+
+  depends_on "cmake" => :build
+
+  def install
+    system "cmake", ".", *std_cmake_args
+    system "make", "install"
+    pkgshare.install "samples/sgemm.c" # for a simple library linking test
```
Comment (fxcoudert) [inline Formula/clblast.rb]: Let's install the whole samples directory

File: Formula/clblast.rb [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+  end
+
+  test do
+    system ENV.cc, pkgshare/"sgemm.c", "-I#{include}", "-L#{lib}",
```
Comment (fxcoudert) [inline Formula/clblast.rb]: And adjust the test (`pkgshare/"samples/sgemm.c"`)

File: Formula/clblast.rb (hunk 4 above)
Comment (CNugteren) [reply Formula/clblast.rb]: OK, I did that. And I adjusted the test to `pkgshare/"samples\/sgemm.c"`, I guess I have to escape the slash?

Comment (DomT4) [reply Formula/clblast.rb]: Not unless it's a regex, which it isn't.

Comment (CNugteren) [reply Formula/clblast.rb]: OK, fixed it.

**Comment by fxcoudert** [comment]
Thanks @CNugteren for the pull request! and congrats on your first contribution to Homebrew

**Pull request closed without merging**